pkg/workqueue/config.go

```diff
@@ -0,0 +1,105 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workqueue
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+const (
+	ItemRateLimiterBaseDelayKey     = "item-rate-limiter-base-delay"
+	ItemRateLimiterMaxDelayKey      = "item-rate-limiter-max-delay"
+	OverallRateLimiterMaxDelayKey   = "overall-rate-limiter-max-delay"
+	BucketRateLimiterItemsPerSecKey = "bucket-rate-limiter-items-per-sec"
+	BucketRateLimiterMaxBurstKey    = "bucket-rate-limiter-max-burst"
+)
+
+type Config struct {
+	ItemRateLimiterBaseDelay     time.Duration
+	ItemRateLimiterMaxDelay      time.Duration
+	OverallRateLimiterMaxDelay   time.Duration
+	BucketRateLimiterItemsPerSec int
+	BucketRateLimiterMaxBurst    int
+}
+
+func DefaultConfig() Config {
+	return Config{
+		ItemRateLimiterBaseDelay:     time.Millisecond,
+		ItemRateLimiterMaxDelay:      30 * time.Second,
+		OverallRateLimiterMaxDelay:   5 * time.Minute,
+		BucketRateLimiterItemsPerSec: 10,
+		BucketRateLimiterMaxBurst:    500,
+	}
+}
+
+func DefaultConfigIfNil(c *Config) Config {
+	if c == nil {
+		return DefaultConfig()
+	}
+
+	return *c
+}
+
+func ConfigFromConfigMap(configMap *corev1.ConfigMap, keyPrefix string, defaultConfig *Config) *Config {
+	if configMap == nil {
+		return defaultConfig
```

Comment:
This is _really_ minor, but perhaps

```suggestion
		return DefaultConfigIfNil(defaultConfig)
```

Reply:
But we've already checked for nil on L66 so we don't need to call `DefaultConfigIfNil`.

Reply:
I didn't do that b/c I assumed the caller probably wants nil returned if they passed nil `defaultConfig`.